docs/gcp.md

```diff
@@ -0,0 +1,205 @@
+
+1. Pull the latest image from the Artifact Registry to retrieve cached layers
+2. Build the image with `$SHORT_SHA` and `latest` tag
+3. Push the image to the Artifact Registry
+4. Deploy the image to Cloud Run
+
+`images` specifies the build images to store in the registry. The default
+`timeout` is 10 minutes, so we'll need to increase it for Swift builds. We use
+`8080` as the default port here, though it's recommended to remove this line and
+have the server listen on `$PORT`.
+
+## Deployment
+
+![cloud build trigger settings and how to connect a code repository](../images/gcp-connect-repo.png)
+
+Push all files to a remote repository. Cloud Build currently supports, GitHub,
+Bitbucket and GitLab. now) and head to
+[Cloud Build Triggers](https://console.cloud.google.com/cloud-build/triggers)
+and click "Create Trigger":
+
+1. Add a name and description
+2. Event: "Push to a branch" is active
+3. Source: "Connect New Repository" and authorize with your code provider, and
+   add the repository where your Swift server code is hosted. Note that you need
+   to configure
+   [GitHub](https://cloud.google.com/build/docs/automating-builds/build-repos-from-github),
+   [GitLab](https://cloud.google.com/build/docs/automating-builds/build-repos-from-gitlab)
+   or
+   [Bitbucket](https://cloud.google.com/build/docs/automating-builds/build-repos-from-bitbucket-cloud)
+   to allow GCP access first.
+4. Configuration: "Cloud Build configuration file" / Location: Repository
+5. Advanced:
+   [Substitution variables](https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values):
+   You need to set the variables for region, repository name and service name
+   here. You can pick a
+   [region of your choice](https://cloud.google.com/about/locations/) (ie.
+   `us-central1`). All custom variables must start with an underscore
+   (`_REGION`). `_REPOSITORY_NAME` and `_SERVICE_NAME` are up to you. If you use
+   environment variables for example to connect to a database or 3rd party
+   services, you can set the values here too.
+6. "Create"
+
+As a last step before deploying the new service, go to the
+[Cloud Build Settings](https://console.cloud.google.com/cloud-build/settings)
+and make sure "Cloud Run" is enabled. This gives Cloud Build the necessary IAM
+permissions to deploy Cloud Run services.
+
+![cloud build settings](../images/gcp-cloud-build-settings.png)
+
+In the Trigger overview page, you should see your new "swift-service" trigger.
+Click on "RUN" on the right to start the trigger manually from the `main`
+branch. With a simple Hummingbird project the build takes about 7-8 minutes.
+Vapor takes about 25 minutes on the standard/small build machines, which are
+fairly slow. "Jordane" from the Vapor Discord community
+[recommends using `machineType: E2_HIGHCPU_8`](https://discord.com/channels/431917998102675485/447893851374616576/915819735738888222)
+in the `cloudbuild.yaml` to speed up deployments:
```

Comment:
do we know if the slowness is about dependency resolution or build / CPU? Is there a way to configure dependencies cache location?

Reply:
That was on my todo list to figure out. Caching is definitely supported on both GCP and AWS, should be possible with a multi-stage build to separate the dependencies.

Reply:
The majority of the time is build time. Both the default build machine and high CPU build machine were about a couple of minutes for resolution (high CPU was about 30s faster) but actual build time for default was 20 mins and about 5 mins on the high CPU machine